crates/rnote-ui/src/appwindow/actions.rs

```diff
@@ -1160,38 +1160,73 @@ impl RnAppWindow {
                 async move {
                     debug!("Recognized clipboard content format: files list");
 
-                    match appwindow.clipboard().read_text_future().await {
-                        Ok(Some(text)) => {
-                            let file_paths = text
-                                .lines()
-                                .filter_map(|line| {
-                                    let file_path = if let Ok(path_uri) = url::Url::parse(line) {
-                                        path_uri.to_file_path().ok()?
-                                    } else {
-                                        PathBuf::from(&line)
-                                    };
-
-                                    if file_path.exists() {
-                                        Some(file_path)
-                                    } else {
-                                        None
-                                    }
-                                })
-                                .collect::<Vec<PathBuf>>();
-
-                            for file_path in file_paths {
-                                appwindow
-                                    .open_file_w_dialogs(
-                                        gio::File::for_path(&file_path),
-                                        target_pos,
-                                        true,
+                    match appwindow
```

Comment:
This is a candidate for DRY - almost the same code is used in three places in this source file

Reply:
Fixed